packages/regl-worldview/src/Worldview.js

```diff
@@ -197,41 +202,58 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
     }
   }
 
-  _onDoubleClick = (e: SyntheticMouseEvent<HTMLCanvasElement>) => {
-    this._onMouseInteraction(e, "onDoubleClick");
+  handleOffscreenMouseEvent = (e: SyntheticMouseEvent<HTMLCanvasElement>, mouseEventName: MouseEventEnum) => {
+    if (mouseEventName === "onDoubleClick") {
+      this._onDoubleClick(e, true);
+    } else if (mouseEventName === "onMouseDown") {
+      this._onMouseDown(e, true);
+    } else if (mouseEventName === "onMouseMove") {
+      this._onMouseMove(e, true);
+    } else if (mouseEventName === "onMouseUp") {
+      this._onMouseUp(e, true);
+    }
+  };
+
+  _onDoubleClick = (e: SyntheticMouseEvent<HTMLCanvasElement>, fromOffscreenTarget: boolean) => {
+    this._onMouseInteraction(e, "onDoubleClick", !!fromOffscreenTarget);
   };
 
-  _onMouseDown = (e: SyntheticMouseEvent<HTMLCanvasElement>) => {
+  _onMouseDown = (e: SyntheticMouseEvent<HTMLCanvasElement>, fromOffscreenTarget: boolean) => {
     this._dragStartPos = { x: e.clientX, y: e.clientY };
-    this._onMouseInteraction(e, "onMouseDown");
+    this._onMouseInteraction(e, "onMouseDown", !!fromOffscreenTarget);
   };
 
-  _onMouseMove = (e: SyntheticMouseEvent<HTMLCanvasElement>) => {
-    this._onMouseInteraction(e, "onMouseMove");
+  _onMouseMove = (e: SyntheticMouseEvent<HTMLCanvasElement>, fromOffscreenTarget: boolean) => {
+    this._onMouseInteraction(e, "onMouseMove", !!fromOffscreenTarget);
   };
 
-  _onMouseUp = (e: SyntheticMouseEvent<HTMLCanvasElement>) => {
-    this._onMouseInteraction(e, "onMouseUp");
+  _onMouseUp = (e: SyntheticMouseEvent<HTMLCanvasElement>, fromOffscreenTarget: boolean) => {
+    this._onMouseInteraction(e, "onMouseUp", !!fromOffscreenTarget);
     const { _dragStartPos } = this;
     if (_dragStartPos) {
       const deltaX = e.clientX - _dragStartPos.x;
       const deltaY = e.clientY - _dragStartPos.y;
       const distance = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
       if (distance < DEFAULT_MOUSE_CLICK_RADIUS) {
-        this._onMouseInteraction(e, "onClick");
+        this._onMouseInteraction(e, "onClick", !!fromOffscreenTarget);
       }
       this._dragStartPos = null;
     }
   };
 
-  _onMouseInteraction = (e: SyntheticMouseEvent<HTMLCanvasElement>, mouseEventName: MouseEventEnum) => {
+  _onMouseInteraction = (
+    e: SyntheticMouseEvent<HTMLCanvasElement>,
+    mouseEventName: MouseEventEnum,
+    fromOffscreenTarget: boolean
+  ) => {
     const { worldviewContext } = this.state;
     const worldviewHandler = this.props[mouseEventName];
 
-    if (!(e.target instanceof window.HTMLElement) || e.button !== 0) {
+    if (!fromOffscreenTarget && (!(e.target instanceof window.HTMLElement) || e.button !== 0)) {
```

Comment:
Worth a comment saying why these checks aren't needed when the canvas is provided I think (or why they _are_ needed when the canvas isn't provided, if we know.)

Reply:
The check is required because `window` is not defined for a web worker and the check throws an error. Checking for `fromOffscreenTarget` first prevents the error and also ignores the actual type of `e.target`, which is not an actual `HTMLElement`.

Reply:
Makes sense. Still, I think it's worth a comment in the code -- maybe just "assume events coming from handleOffscreenMouseEvent have relevant targets".